lib/ansible/modules/notification/mail.py

```diff
@@ -255,6 +255,8 @@ def main():
                 if secure == 'always':
                     module.fail_json(rc=1, msg='Unable to start an encrypted session to %s:%s: %s' %
                                                (host, port, to_native(e)), exception=traceback.format_exc())
+            except:
+                pass
```

Comment:
I'd do some debug logging in here. Swallowing wholesale exceptions like this is harmful. Let's at least do `except Exception as exc:` and add log that + I'd like to see an explicit section for that socket error as well.

Reply:
The whole purpose here is that it falls back to the second attempt, except if `secure == always`.

Reply:
Yeah, but it's untrackable this way. Also, there's too much nesting in here. It's hard to reason about it. It's better for maintainability to set explicit expectations for what we want to ignore.

Reply:
We should never use a bare except in code.  except Exception is acceptable when it is a catchall case (like this).  bare except catches things like KeyboardInterrupt and SystemExit which we do not want to catch in 99% of circumstances.  In the other 1%, we should be explicit about catching those exceptions so that people can tell we know what we're doing.  (Changing to except Exception isn't sufficient to block this as a backport, though.  I'm going to merge this and the change can be made in devel).